### PR TITLE
Feature/liquity staking actions

### PR DESCRIPTION
**Pull request by kjaredic**

### CODE REVIEW

File: contracts/actions/liquity/LiquityStake.sol [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            _params.lqtyAmount = LQTYTokenAddr.getBalance(_params.from);
+        }
+
+        uint256 ethGain = LQTYStaking.getPendingETHGain(address(this));
+        uint256 lusdGain = LQTYStaking.getPendingLUSDGain(address(this));
+
+        LQTYTokenAddr.pullTokensIfNeeded(_params.from, _params.lqtyAmount);
+        LQTYStaking.stake(_params.lqtyAmount);
+
+        TokenUtils.depositWeth(ethGain);
```
Comment (yippee-ki-yay) [inline contracts/actions/liquity/LiquityStake.sol:63]: This pattern is used in more than 2 places, create a separate function in helper that will send the eth/lusd gains

File: contracts/actions/liquity/LiquityStake.sol [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    function executeAction(
+        bytes[] memory _callData,
+        bytes[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override returns (bytes32) {
+        Params memory params = parseInputs(_callData);
+        params.lqtyAmount = _parseParamUint(params.lqtyAmount, _paramMapping[0], _subData, _returnValues);
+        params.from = _parseParamAddr(params.from, _paramMapping[1], _subData, _returnValues);
+        params.wethTo = _parseParamAddr(params.wethTo, _paramMapping[2], _subData, _returnValues);
```
Comment (yippee-ki-yay) [inline contracts/actions/liquity/LiquityStake.sol:30]: Same as in other actions, can be 1 address

**Review by yippee-ki-yay — CHANGES_REQUESTED** [review]
Few of the same issues as in other recent liquity actions

File: contracts/actions/liquity/LiquityStake.sol (hunk 2 above)
Comment (kjaredic) [reply contracts/actions/liquity/LiquityStake.sol:30]: non issue

**Review by yippee-ki-yay — APPROVED** [review]
lgtm

**Pull request merged**